inc/sitemaps/class-sitemap-image-parser.php

```diff
@@ -425,6 +393,92 @@ protected function get_absolute_url( $src ) {
 	}
 
 	/**
+	 * Returns the attachments for a gallery.
+	 *
+	 * @param int   $id      The post id.
+	 * @param array $gallery The gallery config.
+	 *
+	 * @return array The selected attachments
+	 */
+	protected function get_gallery_attachments( $id, $gallery ) {
+
+		// When there are posts to include.
+		if ( ! empty( $gallery['include'] ) ) {
+			$_attachments = $this->get_attachments(
+				array(
+					'include'     => $gallery['include'],
+					'numberposts' => 5,
+				)
+			);
+
+			$gallery_attachments = array();
+			foreach ( $_attachments as $key => $val ) {
+				$gallery_attachments[ $val->ID ] = $_attachments[ $key ];
+			}
+
+			return $gallery_attachments;
+		}
+
+		// When $id is empty, just return empty array.
+		if ( empty( $id ) ) {
+			return array();
+		}
+
+		return $this->get_gallery_attachments_for_parent( $id, $gallery );
+	}
+
+	/**
+	 * Returns the attachments for the given id.
+	 *
+	 * @param int   $id      The post id.
+	 * @param array $gallery The gallery config.
+	 *
+	 * @return array The selected attachments.
+	 */
+	protected function get_gallery_attachments_for_parent( $id, $gallery ) {
+		$query = array(
+			'numberposts' => -1,
+			'post_parent' => $id,
+		);
+
+		// When there are posts that should be excluded from result set.
+		if ( ! empty( $gallery['exclude'] ) ) {
+			$query['exclude'] = $gallery['exclude'];
+		}
+
+		return $this->get_attachments( $query );
+	}
+
+	/**
+	 * Returns the attachments.
+	 *
+	 * @param array $args Array with query args.
+	 *
+	 * @return array The found attachments.
+	 */
+	protected function get_attachments( $args ) {
+		$default_args = array(
+			'post_status'      => 'inherit',
+			'post_type'        => 'attachment',
+			'post_mime_type'   => 'image',
+
+			// Defaults taken from function get_posts.
```

Comment:
`get_posts` also has the defaults `$r['ignore_sticky_posts'] = true;` and `$r['no_found_rows'] = true;`.